source/swap/contracts/Swap.sol

```diff
@@ -335,10 +339,13 @@ contract Swap is ISwap {
   ) internal {
     if (kind == ERC721_INTERFACE_ID) {
       // Attempt to transfer an ERC-721 token.
-      IERC721(token).safeTransferFrom(from, to, param);
+      IERC721(token).transferFrom(from, to, param);
     } else {
-      // Attempt to transfer an ERC-20 token.
-      require(IERC20(token).transferFrom(from, to, param));
+      require(from != to, "TO_CANNOT_EQUAL_FROM");
```

Comment:
Should this check be taken out the front of the if statement? Does it make sense to have IERC721 transferring tokens with from and to the same?

Reply:
I can move it up, just initially was trying to keep it contained.